tools/Bridge/Bridge/IdleTimeoutManager.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http;
+
+namespace Bridge
+{
+    public class IdleTimeoutHandler : DelegatingHandler
+    {
+        IdleTimeoutManager _timeoutManager;
+
+        private IdleTimeoutHandler(TimeSpan idleTimeout)
+        {
+            _timeoutManager = new IdleTimeoutManager(idleTimeout); ;
+            _timeoutManager.OnTimeOut += (s, e) =>
+            {
+                Console.WriteLine("Timed out as there were no messages to the bridge for {0} seconds", (int)e.TotalSeconds);
+                Environment.Exit(-1);
```

Comment:
Hmm thought about it a bit more and decided to avoid adding a file for a single constant. 
I think we will have to consolidate exit codes when we have more than one.